File: backend/src/main/kotlin/com/localbridge/backend/entity/InspectionItem.kt
package com.localbridge.backend.entity

import jakarta.persistence.*
import java.time.Instant

@Entity
@Table(name = "inspection_items")
data class InspectionItem(
    @Id
    @Column(length = 36)
    val id: String,

    @Column(name = "inspection_id", nullable = false, length = 36)
    val inspectionId: String,

    @Column(nullable = false, length = 255)
    val title: String,

    @Column(columnDefinition = "TEXT")
    val description: String? = null,

    @Column(name = "area_id", nullable = false, length = 36)
    val areaId: String,

    @Column(name = "equipment_id", nullable = false, length = 36)
    val equipmentId: String,

    @Column(nullable = false, length = 50)
    @Enumerated(EnumType.STRING)
    val status: InspectionStatus,

    @Column(name = "created_at", nullable = false)
    val createdAt: Instant = Instant.now(),

    @Column(name = "updated_at", nullable = false)
    val updatedAt: Instant = Instant.now()
)

File: frontend/src/domain/entities/InspectionItem.ts
import type { InspectionStatus } from '@/domain/types/inspection'

export class InspectionItem {
  readonly id: string
  readonly inspectionId: string
  readonly title: string
  readonly areaId: string
  readonly equipmentId: string
  status: InspectionStatus
  readonly createdAt: number
  updatedAt: number
  readonly description?: string

  constructor(
    id: string,
    inspectionId: string,
    title: string,
    areaId: string,
    equipmentId: string,
    status: InspectionStatus,
    createdAt: number,
    updatedAt: number,
    description?: string
  ) {
    this.id = id
    this.inspectionId = inspectionId
    this.title = title
    this.areaId = areaId
    this.equipmentId = equipmentId
    this.status = status
    this.createdAt = createdAt
    this.updatedAt = updatedAt
    this.description = description
  }
}
